### DISCUSSION TITLE
Print quarto version in the generated document both   in command line, knitr and jupyter generated document

### DISCUSSION THREAD

**Discussion by danieltomasz** [quarto-dev/quarto-cli#11349] [Q&A]
### Description

I am helping to develop some templates that need to support some feature between different quarto version, for the testing purpose how I can output/print quarto version the document was generated?
For quarto in Rstudio I can use
````
```{r} 
quarto::quarto_version()
```
````
but when I run test from command line (no R or without this library it will fail) but I might give me false results  when I set in command line different path than the one shipped with Rstudio (though I am not so sure about this)
Is there any way to do it consistently?

**Comment by cscheid** [comment]
In an HTML document you always have a `meta` element. From quarto.org:

<img width="535" alt="image" src="https://github.com/user-attachments/assets/e9d4210f-ca32-4d47-a9a6-48de3034be91">

What formats are you targeting?

**Reply by danieltomasz** [reply]
its a pdf via `typst` template (easiest way is to install python/R and  do this via chunk, bash call might be not OS portable)

**Reply by mcanouil** [reply]
See <https://github.com/mcanouil/quarto-lua-env>.

<table>
<tr><th>Input</th><th>Output</th></tr>
<tr><td>

````qmd
---
title: "Quarto Playground"
format: html
filters:
  - lua-env
---

This is a playground for Quarto {{< lua-env quarto.version >}}.

````

</td><td>

<img width="466" alt="image" src="https://github.com/user-attachments/assets/15a0d984-f543-48f4-8f35-5a4d952dc325">

</td></tr>
</table>

**Reply by mcanouil** [reply]
Derived from the above:

```lua
return {
  ['qv'] = function(args, kwargs, meta)
    return table.concat(quarto.version, '.')
  end
}
```

```qmd
---
title: "Quarto Playground"
format: html
shortcodes: 
  - version.lua
---

This is a playground for Quarto {{< qv >}}.
````

**Comment by cderv** [comment]
There is also `QUARTO_VERSION` exported by Quarto as a major.minor so you could do 

````
---
title: "Quarto Playground"
format: typst
---

This is a playground for Quarto {{< env QUARTO_VERSION >}}.
````

![image](https://github.com/user-attachments/assets/f73a306e-affc-46c0-ba85-02d3dc98d83c)

But the [lua filter above ](https://github.com/quarto-dev/quarto-cli/discussions/11349#discussioncomment-11215849) for shortcode is a really good way.
